query-engine/driver-adapters/src/transaction.rs

```diff
@@ -6,10 +8,78 @@ use quaint::{
     Value,
 };
 
-use crate::proxy::{TransactionOptions, TransactionProxy};
+use crate::proxy::{TransactionContextProxy, TransactionOptions, TransactionProxy};
 use crate::{proxy::CommonProxy, queryable::JsBaseQueryable, send_future::UnsafeFuture};
 use crate::{JsObject, JsResult};
 
+pub(crate) struct JsTransactionContext {
+    tx_ctx_proxy: TransactionContextProxy,
+    inner: JsBaseQueryable,
+}
+
+// Wrapper around JS transaction context objects that implements Queryable. Can be used in place of quaint transaction,
+// context, but delegates most operations to JS
+impl JsTransactionContext {
+    pub(crate) fn new(inner: JsBaseQueryable, tx_ctx_proxy: TransactionContextProxy) -> Self {
+        Self { inner, tx_ctx_proxy }
+    }
+
+    pub fn start_transaction(&self) -> UnsafeFuture<impl Future<Output = quaint::Result<Box<JsTransaction>>> + '_> {
+        UnsafeFuture(self.tx_ctx_proxy.start_transaction())
+    }
+}
+
+#[async_trait]
+impl Queryable for JsTransactionContext {
```

Comment:
We need `JsTransactionContext` to be `Queryable` because of its dependency on the Queryable methods `requires_isolation_first` and `set_tx_isolation_level`, which in turn depends on `raw_cmd`. If you attempt removing the `Queryable` dependency on `JsTransactionContext`, you end up getting errors like

```sh
node[34371]: ../src/node_api.cc:1395:napi_status napi_release_threadsafe_function(napi_threadsafe_function, napi_threadsafe_function_release_mode): Assertion `(func) != nullptr' failed.
 1: 0x100ad2114 node::Abort() [/Users/jkomyno/.nvm/versions/node/v20.9.0/bin/node]
 2: 0x100ad1e64 node::PrintCaughtException(v8::Isolate*, v8::Local<v8::Context>, v8::TryCatch const&) [/Users/jkomyno/.nvm/versions/node/v20.9.0/bin/node]
```